ngmix/prepsfmom.py

```diff
@@ -0,0 +1,639 @@
+import logging
+
+import numpy as np
+import scipy.fft as fft
+
+from ngmix.observation import Observation
+from ngmix.moments import fwhm_to_sigma
+from ngmix.util import get_ratio_error
+from ngmix.gexceptions import FFTRangeError
+from ngmix.fastexp_nb import FASTEXP_MAX_CHI2, fexp_arr
+
+
+logger = logging.getLogger(__name__)
+
+
+class _PrePSFMom(object):
+    """Measure pre-PSF weighted real-space moments.
+
+    This class is not meant to be used directly. Instead use either `KSigmaMom`
+    or `PrePSFGaussMom`.
+
+    If the fwhm of the weight/kernel function is of similar size to the PSF or
+    smaller, then the object properties returned by this fitter will be very noisy.
+
+    Parameters
+    ----------
+    fwhm : float
+        This parameter is the approximate real-space FWHM of the kernel. The units are
+        whatever the Jacobian on the obs converts pixels units to. This is typically
+        arcseconds.
+    kernel : str
+        The kernel to use. Either `ksigma` or `gauss`.
+    pad_factor : int, optional
+        The factor by which to pad the FFTs used for the image. Default is 4.
+    """
+    def __init__(self, fwhm, *, kernel, pad_factor=4):
```

Comment:
We debated this a while back and decided to avoid `*` in the ngmix code because 1) no ngmix code uses it, and 2) lthe user can always use keywords by choice

Reply:
fixed